### PR TITLE
fix: Avoid applying ELF-specific entry behaviour to other platforms

### PR DESCRIPTION
**Pull request by davidlattimore**
For ELF, we duplicate some of the dubious behaviours of GNU ld - e.g. if the entry symbol isn't defined, we warn, then use the start of the .text section or otherwise 0. We don't want to duplicate such behaviours to other platforms, so this PR moves this behaviour to be ELF-specific. In order to properly test the entry behaviour of the other platforms I needed to implement the `-e` flag for mac. A bunch of new tests were added to check the corner cases around entry handlings. Many of these fail without the rest of the change.

### CODE REVIEW

File: libwild/src/elf_writer.rs [hunk 1]
```diff
@@ -486,6 +486,40 @@ fn populate_file_header<A: Arch<Platform = Elf>>(
...
+    let text_layout = layout.section_layouts.get(output_section_id::TEXT);
+    if text_layout.mem_size == 0 {
+        layout.symbol_db.warning(format!(
+            "cannot find entry symbol `{entry_name}` and .text is empty, not setting entry point"
+        ));
+        return Ok(0);
+    }
+
+    layout.symbol_db.warning(format!(
+        "cannot find entry symbol `{entry_name}`, defaulting to 0x{}",
```
Comment (lapla-cogito) [inline libwild/src/elf_writer.rs]:
```suggestion
        "cannot find entry symbol `{entry_name}`, defaulting to 0x{:x}",
```
